libraries/Microsoft.Bot.Builder/BotAdapter.cs

```diff
@@ -224,6 +225,22 @@ protected async Task RunPipelineAsync(ITurnContext turnContext, BotCallbackHandl
             // Call any registered Middleware Components looking for ReceiveActivityAsync()
             if (turnContext.Activity != null)
             {
+                if (turnContext.Activity.Locale != null)
+                {
+                    try
+                    {
+                        Thread.CurrentThread.CurrentCulture = new CultureInfo(turnContext.Activity.Locale);
+                        (turnContext as TurnContext).Locale = turnContext.Activity.Locale;
+                    }
+#pragma warning disable CA1031 // Do not catch general exception types
+                    catch
```

Comment:
Refactor this, do not catch all exceptions, in this case it looks like you are trying to catch an exception that may occur if you send an invalid locale code and try to create a cultureInfo instance with that. Catch that specific exception instead and do not add the pragma.

Reply:
change to CultureNotFoundException